fuzz/Cargo.toml

```diff
@@ -24,3 +24,9 @@ name = "mutation"
 path = "fuzz_targets/mutation.rs"
 test = false
 doc = false
+
+[[bin]]
+name = "hash"
+path = "fuzz_targets/hash.rs"
+test = false
+doc = false
```

Comment:
I see that this tests makes more sense as a proptests. II implemented as a fuzz tests because property tests often don't run long enough/can't use debug information to exhaust all possible code paths (and there is really no practical difference between them except a nicer reduce logic apart from that). I didn't change this yet because you said you will reconsider below. Please just let me know if I should turn this into a propertytest or not and I will do so